test/files/scalacheck/MutableTreeSet.scala

```diff
@@ -0,0 +1,200 @@
+import java.io._
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.util.Try
+
+package scala.collection.mutable {
+
+  object MutableTreeSetProperties extends Properties("mutable.TreeSet") {
+    type K = String
+
+    property("size, isEmpty") = forAll { (elems: Set[K]) =>
+      val set = mutable.TreeSet[K]()
+      set ++= elems
+      set.size == elems.size && set.isEmpty == elems.isEmpty
+    }
+
+    property("+=") = forAll { (set: mutable.TreeSet[K], k: K) =>
+      val oldSize = set.size
+      val containedKeyBefore = set.contains(k)
+      val newExpectedSize = if(containedKeyBefore) oldSize else oldSize + 1
+
+      set += k
+      set.contains(k) && set.size == newExpectedSize
+    }
+
+    property("++=") = forAll { (set: mutable.TreeSet[K], ks: Seq[K]) =>
+      set ++= ks
+      ks.toSet.forall(set.contains)
+    }
+
+    property("-=") = forAll { (set: mutable.TreeSet[K], k: K) =>
+      val oldSize = set.size
+      val containedKeyBefore = set.contains(k)
+      val newExpectedSize = if(containedKeyBefore) oldSize - 1 else oldSize
+
+      set -= k
+      !set.contains(k) && set.size == newExpectedSize
+    }
+
+    property("--=") = forAll { (set: mutable.TreeSet[K], ks: Seq[K]) =>
+      set --= ks
+      ks.toSet.forall(!set.contains(_))
```

Comment:
You should additionally check here that the `set` contains all the elements that were in the original set, but are not in the sequence.

Btw, in case of an error, how does this play with ScalaCheck's minimization mechanism? IIRC, the same mutable set instance is reused during minimization, which could result in weird error messages. The solution is to use immutable arguments to `forAll`.